flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerServicesConfiguration.java

```diff
@@ -270,7 +271,8 @@ public static TaskManagerServicesConfiguration fromConfiguration(
 
 		final String[] alwaysParentFirstLoaderPatterns = CoreOptions.getParentFirstLoaderPatterns(configuration);
 
-		final int numIoThreads = configuration.get(TaskManagerOptions.NUM_IO_THREADS);
+		// multiply core-count to be on the safer side, since we used a pool with size=64 in the past
+		final int numIoThreads = ClusterEntrypointUtils.getPoolSize(configuration) * 4;
```

Comment:
```suggestion
		final int numIoThreads = ClusterEntrypointUtils.getPoolSize(configuration);
```

I will update `getPoolSize` to return the new default value of `4 * cores` as part of changing the type of the thread pool on the JM side.